components/oauth/org.wso2.carbon.identity.oauth.ui/src/main/java/org/wso2/carbon/identity/oauth/ui/OAuthServlet.java

```diff
@@ -117,7 +117,7 @@ protected void service(HttpServletRequest req, HttpServletResponse resp)
                 token = client.getAccessToken(params);
                 accessToken = OAuthConstants.OAUTH_TOKEN + "=" + token.getOauthToken() + "&"
                         + OAuthConstants.OAUTH_TOKEN_SECRET + "=" + token.getOauthTokenSecret();
-                out.write(accessToken);
+                out.write(Encode.forUriComponent(accessToken));
```

Comment:
It's wrong to encode the full query string for percent encoding. That might break the uri. Instead encode only the untrusted query parameter values. That is token.getOauthToken() and token.getOauthTokenSecret()